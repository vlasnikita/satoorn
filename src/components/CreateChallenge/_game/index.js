import React, {Component, Fragment} from 'react';
import {connect} from "react-redux";

import CreateChallenge from "Components/CreateChallenge";
import {setActiveStreamer} from "Actions/streamers";
import {setCreateChallengeDescription, setCreateChallengeDraft, setCreateChallengeTitle} from "Actions/createChallenge";
import {setRedirectToChallenge} from "Actions/challenges";
import {addActiveTransaction} from "Actions/transactions";
import {openPopup} from "Actions/popup";
import {showAlert} from "Actions/alert";

export class CreateChallengeGame extends CreateChallenge {

    constructor(props) {
        super(props);

        this.state = {
            isErrors: [],
            newBidAmount: props.configs.minPaymentAmount.value,
            minBid: props.configs.minPaymentAmount.value,
            isInputFocused: false
        }
    }

    render() {
        return (
            <div className="Island CreateChallenge_game">
                <div className="CreateChallenge__row">
                    <span className="CreateChallenge__label">Название</span>
                    <input
                        autoComplete="off"
                        name='title'
                        id='title'
                        type='text'
                        className={`CreateChallenge__input ${this.state.isErrors.includes('title') && 'CreateChallenge__input_error'}`}
                        value={this.state.title}
                        onChange={e => {
                            this.handleSetCreateChallengeTitle(e.target.value)
                            this.resetInputErrors(e)
                        }}
                    />
                    {this.state.isErrors.includes('title') &&
                    <p className="CreateChallenge__input_errorMessage">Не указана игра</p>}
                </div>
                <div className="CreateChallenge__row">
                    {this.getActions()}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    profile: state.profile.profile,
    configs: state.configs,

    templateDescription: state.createChallenge.description,
    challengeDraft: state.createChallenge.challengeDraft,

    activeStreamer: state.activeStreamer.activeStreamer,
    isActiveStreamerLoading: state.activeStreamer.loading,
    isActiveStreamerLoaded: state.activeStreamer.loaded,

    activeTransaction: state.transactions.activeTransaction,
    redirectToChallenge: state.challenges.redirectToChallenge
});

const mapDispatchToProps = (dispatch) => ({
    handleSetActiveStreamer: name => dispatch(setActiveStreamer(name)),

    updateSavedChallengeTitle: title => dispatch(setCreateChallengeTitle(title)),
    updateSavedChallengeDescription: description => dispatch(setCreateChallengeDescription(description)),
    handleResetCreateChallengeDraft: () => dispatch(setCreateChallengeDraft(null)),

    handleSetRedirectToChallenge: challengeId => dispatch(setRedirectToChallenge(challengeId)),
    handleAddActiveTransaction: transaction => dispatch(addActiveTransaction(transaction)),

    handleOpenPopup: (type, acceptingChallenge) => dispatch(openPopup(type, acceptingChallenge)),
    handleShowAlert: type => dispatch(showAlert(type))
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateChallengeGame)
