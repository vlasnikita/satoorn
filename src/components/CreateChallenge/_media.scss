@media only screen
and (max-width : 1300px){
  .CreateChallenge {
    &__body {
      max-width: 500px;
      margin-top: 30px;
    }
  }
}

@media only screen
and (max-width : 1150px){
  .CreateChallenge {
    &__body {
      max-width: 400px;
      margin-right: 20px;
    }
  }
}

@media only screen
and (max-width : 800px){
  .CreateChallenge {
    border-top: 2px solid #ddd;

    &__body {
      width: 100%;
      max-width: unset;
      height: calc(100% - 32px);
      margin: 0;
      border-radius: 0;
    }

    &__button{
      top: unset;
      bottom: 0;
      margin-top: 0;
      margin-bottom: 10px;
    }
  }
}

@media only screen
and (max-width : 674px){
  .CreateChallenge{
    height: calc(100% - 120px);
  }
}

@media only screen
and (max-width : 650px){
  .CreateChallenge {

    &__button{
      top: unset;
      bottom: 0;
      margin-top: 0;
      margin-bottom: 10px;
    }
  }
}