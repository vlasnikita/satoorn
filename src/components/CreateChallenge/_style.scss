.CreateChallenge {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  height: 100%;
  background: #fff;

  &__body{
    max-width: 600px;
    background: #fff;
    height: auto;
    border-radius: 8px;
    position: relative;
    margin-top: 40px;
  }

  .Challenge__actions {
    height: 32px;
  }
}

.CreateChallenge__row {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  margin-bottom: 40px;

  &_payment{
    width: 240px;
  }
}

.CreateChallenge__label{
  font-size: 1.2rem;
  color: #222;
  margin-bottom: 4px;
  font-weight: bold;
}

.CreateChallenge__input{
  position: relative;
  border: 1px solid #ddd;
  padding: 10px;
  border-radius: 4px;
  width: calc(100% - 20px);
  font-size: 1.6rem;
  color: #222;
  outline: none;
  transition: .5s;

  &Container{
    width: 46%;
    display: flex;
    position: relative;
    justify-content: center;
    align-items: center;

    border: 1px solid #ddd;
    flex-direction: row;
    height: 44px;
    border-radius: 4px;
    background: rgba(238,238,238,.25);
  }

  &_payment{
    height: 36px;
    border: none;
    background: none;
    outline: none;
    font-size: 2.2rem;
    font-weight: bold;
    text-align: center;
    width: 49%;
    color: #222;
  }

  &_errorMessage{
    display: none;
    height: 20px;
    background: rgba(180,64,64,.5);
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    position: absolute;
    top: 100%;
    left: 0;
    width: auto;
    border: 1px solid #b44040;
    border-top: none;
    padding: 0 5px;
    justify-content: center;
    align-items: center;
    font-size: 0.9rem;
    color: #222;
    font-weight: bold;
  }

  &_error{
    border-color: #b44040 !important;
    border-bottom-left-radius: 0 !important;

    &Message{
      display: flex;
    }
  }

  &_template {
    background: #333;

    &_clicked{
      background: none;
    }
  }
}

.CreateChallenge__action{
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: calc(100% + 10px);
  padding: 0 5px;
  align-items: center;

}

.CreateChallenge__button{
  position: absolute;
  top: 100%;
  margin-top: 10px;
}

.CreateChallenge_game {
  flex-direction: row;
  align-items: center;
  margin: 0;
  margin-bottom: 20px;
  max-width: unset;
  width: calc(100% - 40px);
  height: 90px;
  padding-top: 0;
  padding-bottom: 0;

  .CreateChallenge__row{
    margin-bottom: 0;
  }

  .Challenge__actions{
    margin-top: 0;
    width: 220px;
    margin-left: auto;
  }

  .CreateChallenge__input{
    padding: 6px;
  }

  .CreateChallenge__label{
    position: absolute;
    z-index: 1;
    bottom: 100%;
  }
}

@import "/media";