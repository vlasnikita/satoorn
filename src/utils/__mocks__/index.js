const utils = jest.requireActual('Utils/index')

export const getOperationHash = jest.fn().mockImplementation(str => 'MOCKED_HASH')

export const getDonatersEmoji = jest.fn().mockImplementation(() => utils.getDonatersEmoji())

export const linkifyPlainText = jest.fn().mockImplementation((text) => utils.linkifyPlainText(text))

export const openPaymentWidget =  jest.fn().mockImplementation((amount, transactionId, desc, isDonation, challengeId, challengeBody) =>{})