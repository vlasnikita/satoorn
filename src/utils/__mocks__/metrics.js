export const logAuthButtonClick = jest.fn().mockImplementation(event => {});

export const logDonateEvent = jest.fn().mockImplementation((context, userId) => {})

export const logShareEvent = jest.fn().mockImplementation((context, type, userId) => {})

export const logUserAction = jest.fn().mockImplementation((context, action, userId) => {})