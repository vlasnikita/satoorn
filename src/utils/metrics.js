const amplitude = require('amplitude-js');
const API_KEY = process.env.AMPL_API_KEY;
const loggingEnabled = process.env.NODE_ENV === 'production';

if (loggingEnabled) {
    amplitude.getInstance().init(API_KEY);
}

export const logUserAction = (context, action, userId) => {
    defaultLogEvent(`${context}_${action}_CLICKED`, userId)
}

export const logShareEvent = (context, type, userId) => {
    defaultLogEvent(context + '_SHARE_' + type, userId)
}

export const logDonateEvent = (context, userId) => {
    defaultLogEvent(context + '_DONATE_BUTTON_CLICKED', userId)
}

export const logAuthButtonClick = (event) => {
    defaultLogEvent(event + '_LOGIN_BUTTON_CLICKED', null)
}

export const defaultLogEvent = (event,userId) => {
    if (loggingEnabled) {
        const amplitudeClient = amplitude.getInstance();
        amplitudeClient.setUserId(userId)
        amplitudeClient.logEvent(event)
    }

}
