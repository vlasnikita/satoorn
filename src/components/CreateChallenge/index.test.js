import { configure, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-specific-snapshot';
import React from 'react';
import { Provider } from "react-redux";
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from "redux-mock-store";
import CreateChallenge from './index';
import {defaultTestInitialState, createInputEvent, completePromises} from '../../test/utils/commonTestUtils';

jest.mock('Utils/metrics')
jest.mock('Utils/index')
import {openPaymentWidget} from 'Utils/index';
import { logUserAction } from 'Utils/metrics';

configure({ adapter: new Adapter() })

const mockStore = configureMockStore();

beforeEach(() => {
    fetch.resetMocks();
});

it('renders correctly', () => {
    expect(toJson(renderDefaultCreateChallengeComponent())).toMatchSnapshot();
});

it('should log create challenge event with CHALLENGE_PAGE context and given userId', () => {
    const createChallenge = renderDefaultCreateChallengeComponent()
    createChallenge.find('#CreateChallenge__button').simulate('click')
    expect(logUserAction)
        .toHaveBeenCalledWith('CREATE_CHALLENGE_PAGE', 'CREATE_CHALLENGE', defaultTestInitialState.profile.profile.id)
})


it('should open payment widget with valid input arguments when creates challenge', async () => {
    const testChallengeId = "someChallengeId";
    const testTransactionId = "someTransactionId";
    const testAmount = "500"
    const testTitle = "some challenge title";

    fetch.once(JSON.stringify(
         {
            challengeId: testChallengeId,
            code: "OK",
            transactionId: testTransactionId
          }
    ));


    const createChallenge = renderDefaultCreateChallengeComponent(
        mockStore(defaultTestInitialState)
    )

    createChallenge.find("#title").simulate('change', createInputEvent(testTitle))
    createChallenge.find("#description").simulate('change', createInputEvent("some challenge description"))
    createChallenge.find("#newBidAmount").simulate('change', createInputEvent(testAmount))
    
    await completePromises()
    
    createChallenge.find('#CreateChallenge__button').simulate('click')  

    await completePromises()

    expect(openPaymentWidget).toHaveBeenCalledWith(
        testAmount, 
        testTransactionId,
        `Создание челленджа ${testTitle}`,
        false,
        testChallengeId,
        {donationAmount: testAmount},
        'vlasnikita'
    )
})

function renderDefaultCreateChallengeComponent(store = mockStore(defaultTestInitialState)) {
    return mount(
        <BrowserRouter>
            <Provider store={store}>
                <CreateChallenge />
            </Provider>
        </BrowserRouter>
    );
}

