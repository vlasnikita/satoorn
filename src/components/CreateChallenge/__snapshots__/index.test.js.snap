// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(CreateChallenge))>
        <Route>
          <Connect(CreateChallenge)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <CreateChallenge
              activeStreamer={
                Object {
                  "authenticationSource": "twitch",
                  "bio": null,
                  "challengeCount": 0,
                  "contacts": Array [],
                  "email": null,
                  "externalId": "204865822",
                  "externalPhotoLink": "https://static-cdn.jtvnw.net/user-default-pictures-uv/75305d54-c7cc-40d1-bb9c-91fbe85943c7-profile_image-300x300.png",
                  "externalProfileLink": null,
                  "followerCount": 0,
                  "id": "5f108c57399c8300013dad42",
                  "login": "vlasnikita",
                  "streamer": false,
                  "viewerCount": 0,
                  "walletAmount": 0,
                }
              }
              activeTransaction={
                Object {
                  "id": null,
                  "inProgress": false,
                  "success": false,
                }
              }
              challengeDraft={null}
              configs={
                Object {
                  "minPaymentAmount": Object {
                    "value": 30,
                  },
                  "settings": Object {
                    "loaded": true,
                    "loading": false,
                    "value": Object {
                      "challengeAcceptDetailsConfirmed": true,
                      "challengeDonateDetailsConfirmed": true,
                      "serviceOnboardingConfirmed": true,
                    },
                  },
                }
              }
              handleAddActiveTransaction={[Function]}
              handleOpenPopup={[Function]}
              handleResetCreateChallengeDraft={[Function]}
              handleSetActiveStreamer={[Function]}
              handleSetRedirectToChallenge={[Function]}
              handleShowAlert={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isActiveStreamerLoaded={false}
              isActiveStreamerLoading={false}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              profile={
                Object {
                  "authenticationSource": "twitch",
                  "bio": null,
                  "challengeCount": 0,
                  "contacts": Array [],
                  "email": null,
                  "externalId": "204865822",
                  "externalPhotoLink": "https://static-cdn.jtvnw.net/user-default-pictures-uv/75305d54-c7cc-40d1-bb9c-91fbe85943c7-profile_image-300x300.png",
                  "externalProfileLink": null,
                  "followerCount": 0,
                  "id": "5f108c57399c8300013dad42",
                  "login": "vlasnikita",
                  "streamer": false,
                  "viewerCount": 0,
                  "walletAmount": 0,
                }
              }
              redirectToChallenge={null}
              updateSavedChallengeDescription={[Function]}
              updateSavedChallengeTitle={[Function]}
            >
              <div
                className="Container CreateChallenge"
              >
                <div
                  className="Container__body Island CreateChallenge__body"
                >
                  <div
                    className="CreateChallenge__row"
                  >
                    <span
                      className="CreateChallenge__label"
                    >
                      –ù–∞–∑–≤–∞–Ω–∏–µ
                    </span>
                    <input
                      autoComplete="off"
                      className="CreateChallenge__input false"
                      id="title"
                      name="title"
                      onChange={[Function]}
                      type="text"
                    />
                  </div>
                  <div
                    className="CreateChallenge__row CreateChallenge__row_description"
                  >
                    <span
                      className="CreateChallenge__label"
                    >
                      –û–ø–∏—Å–∞–Ω–∏–µ —á–µ–ª–ª–µ–Ω–¥–∂–∞
                    </span>
                    <textarea
                      className="CreateChallenge__input false"
                      id="description"
                      name="description"
                      onChange={[Function]}
                      rows="5"
                      value=""
                    />
                  </div>
                  <div
                    className="CreateChallenge__row CreateChallenge__row_payment"
                  >
                    <span
                      className="CreateChallenge__label"
                    >
                      –°—É–º–º–∞
                    </span>
                    <div
                      className="Challenge__actions false"
                    >
                      <div
                        className="Challenge__inputContainer"
                      >
                        <input
                          className="Challenge__input Challenge__input_payment"
                          id="newBidAmount"
                          name="newBidAmount"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          type="text"
                          value={30}
                        />
                        <span
                          className="Challenge__inputCurrency Challenge__inputCurrency_2"
                        >
                          ‚ÇΩ
                        </span>
                      </div>
                      <div
                        className="Button Button_purple Challenge__button"
                        id="CreateChallenge__button"
                        onClick={[Function]}
                      >
                        –°–æ–∑–¥–∞—Ç—å
                      </div>
                    </div>
                  </div>
                  <button
                    className="Button Button_greytext CreateChallenge__button"
                    onClick={[Function]}
                  >
                    –û—Ç–º–µ–Ω–∏—Ç—å
                  </button>
                </div>
                <Connect(Templates)>
                  <Templates
                    handleSetCreateChallengeDescription={[Function]}
                    handleSetCreateChallengeType={[Function]}
                    templates={
                      Array [
                        Object {
                          "description": "–ø—Ä–æ—Å–º–æ—Ç—Ä –∞–Ω–∏–º–µ/—Ñ–∏–ª—å–º–∞/—é—Ç—É–±–∞ üçø",
                          "title": "–ø—Ä–æ—Å–º–æ—Ç—Ä –∞–Ω–∏–º–µ/—Ñ–∏–ª—å–º–∞/—é—Ç—É–±–∞ üçø",
                        },
                        Object {
                          "description": "–æ—Ü–µ–Ω–∏—Ç—å –ø–∞–±–ª–∏–∫ –≤–∫",
                          "title": "–æ—Ü–µ–Ω–∏—Ç—å –ø–∞–±–ª–∏–∫ –≤–∫",
                        },
                        Object {
                          "description": "–∫–∞—Ä–∞–æ–∫–µ üé§",
                          "title": "–∫–∞—Ä–∞–æ–∫–µ üé§",
                        },
                        Object {
                          "description": "–º—É–∑—ã–∫–∞ –Ω–∞ –∑–∞–∫–∞–∑",
                          "title": "–º—É–∑—ã–∫–∞ –Ω–∞ –∑–∞–∫–∞–∑",
                        },
                        Object {
                          "description": "–∏–≥—Ä–∞ –Ω–∞ –∑–∞–∫–∞–∑ üéÆ",
                          "title": "–∏–≥—Ä–∞ –Ω–∞ –∑–∞–∫–∞–∑ üéÆ",
                        },
                        Object {
                          "description": "—Å–º–∞–π–ª-–º–æ–¥ –≤ —á–∞—Ç–µ",
                          "title": "—Å–º–∞–π–ª-–º–æ–¥ –≤ —á–∞—Ç–µ",
                        },
                        Object {
                          "description": "–∏–≥—Ä–æ–≤–∞—è –ª–æ—Ç–µ—Ä–µ—è üé≤",
                          "title": "–∏–≥—Ä–æ–≤–∞—è –ª–æ—Ç–µ—Ä–µ—è üé≤",
                        },
                        Object {
                          "description": "–æ—Ü–µ–Ω–∏—Ç—å –∏–Ω—Å—Ç–∞–≥—Ä–∞–º",
                          "title": "–æ—Ü–µ–Ω–∏—Ç—å –∏–Ω—Å—Ç–∞–≥—Ä–∞–º",
                        },
                        Object {
                          "description": "—Å—ã–≥—Ä–∞—Ç—å –≤ –º–∞—Ñ–∏—é üî´",
                          "title": "—Å—ã–≥—Ä–∞—Ç—å –≤ –º–∞—Ñ–∏—é üî´",
                        },
                        Object {
                          "description": "Fall Guys",
                          "title": "Fall Guys",
                        },
                        Object {
                          "description": "Among Us",
                          "title": "Among Us",
                        },
                        Object {
                          "description": "–æ—Ö–æ—Ç–∞ –Ω–∞ –∞—á–∏–≤–∫—É üéØ",
                          "title": "–æ—Ö–æ—Ç–∞ –Ω–∞ –∞—á–∏–≤–∫—É üéØ",
                        },
                        Object {
                          "description": "–∏–≥—Ä–æ–≤–æ–π –º–∞—Ä–∞—Ñ–æ–Ω üèÉ",
                          "title": "–∏–≥—Ä–æ–≤–æ–π –º–∞—Ä–∞—Ñ–æ–Ω üèÉ",
                        },
                        Object {
                          "description": "—Å–ø–∏–¥—Ä–∞–Ω",
                          "title": "—Å–ø–∏–¥—Ä–∞–Ω",
                        },
                        Object {
                          "description": "–æ–±—É—á–µ–Ω–∏–µ –∏–≥—Ä–µ üß†",
                          "title": "–æ–±—É—á–µ–Ω–∏–µ –∏–≥—Ä–µ üß†",
                        },
                        Object {
                          "description": "—Ñ—É–¥-—á–µ–ª–ª–µ–Ω–¥–∂ üçï",
                          "title": "—Ñ—É–¥-—á–µ–ª–ª–µ–Ω–¥–∂ üçï",
                        },
                        Object {
                          "description": "—Å–ø–µ—Ç—å/—Å—Ç–∞–Ω—Ü–µ–≤–∞—Ç—å/–Ω–∞—Ä–∏—Å–æ–≤–∞—Ç—å üé®",
                          "title": "—Å–ø–µ—Ç—å/—Å—Ç–∞–Ω—Ü–µ–≤–∞—Ç—å/–Ω–∞—Ä–∏—Å–æ–≤–∞—Ç—å üé®",
                        },
                        Object {
                          "description": "–∫—É–ª–∏–Ω–∞—Ä–Ω—ã–π —Å—Ç—Ä–∏–º üç≥",
                          "title": "–∫—É–ª–∏–Ω–∞—Ä–Ω—ã–π —Å—Ç—Ä–∏–º üç≥",
                        },
                        Object {
                          "description": "GeoGuessr",
                          "title": "GeoGuessr",
                        },
                        Object {
                          "description": "–ø—Ä–æ–π—Ç–∏ –æ–Ω–ª–∞–π–Ω-—Ç–µ—Å—Ç—ã",
                          "title": "–ø—Ä–æ–π—Ç–∏ –æ–Ω–ª–∞–π–Ω-—Ç–µ—Å—Ç—ã",
                        },
                        Object {
                          "description": "—Ç—É—Ä–Ω–∏—Ä —Å –ø–æ–¥–ø–∏—Å—á–∏–∫–∞–º–∏ üèÜ",
                          "title": "—Ç—É—Ä–Ω–∏—Ä —Å –ø–æ–¥–ø–∏—Å—á–∏–∫–∞–º–∏ üèÜ",
                        },
                        Object {
                          "description": "—Ä–æ–∑—ã–≥—Ä—ã—à–∏",
                          "title": "—Ä–æ–∑—ã–≥—Ä—ã—à–∏",
                        },
                        Object {
                          "description": "—Å–æ–≤–º–µ—Å—Ç–Ω–∞—è –∏–≥—Ä–∞ —Å–æ —Å—Ç—Ä–∏–º–µ—Ä–æ–º üëØ",
                          "title": "—Å–æ–≤–º–µ—Å—Ç–Ω–∞—è –∏–≥—Ä–∞ —Å–æ —Å—Ç—Ä–∏–º–µ—Ä–æ–º üëØ",
                        },
                        Object {
                          "description": "–∫–æ—Å–ø–ª–µ–π",
                          "title": "–∫–æ—Å–ø–ª–µ–π",
                        },
                      ]
                    }
                  >
                    <div
                      className="Templates"
                    >
                      <h3
                        className="Templates__header"
                      >
                        –†–µ–∫–æ–º–µ–Ω–¥–∞—Ü–∏–∏
                      </h3>
                      <div
                        className="Templates__feed"
                      >
                        <div
                          className="Templates__template"
                          key="0"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –ø—Ä–æ—Å–º–æ—Ç—Ä –∞–Ω–∏–º–µ/—Ñ–∏–ª—å–º–∞/—é—Ç—É–±–∞ üçø
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="1"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –æ—Ü–µ–Ω–∏—Ç—å –ø–∞–±–ª–∏–∫ –≤–∫
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="2"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∫–∞—Ä–∞–æ–∫–µ üé§
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="3"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –º—É–∑—ã–∫–∞ –Ω–∞ –∑–∞–∫–∞–∑
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="4"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∏–≥—Ä–∞ –Ω–∞ –∑–∞–∫–∞–∑ üéÆ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="5"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Å–º–∞–π–ª-–º–æ–¥ –≤ —á–∞—Ç–µ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="6"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∏–≥—Ä–æ–≤–∞—è –ª–æ—Ç–µ—Ä–µ—è üé≤
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="7"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –æ—Ü–µ–Ω–∏—Ç—å –∏–Ω—Å—Ç–∞–≥—Ä–∞–º
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="8"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Å—ã–≥—Ä–∞—Ç—å –≤ –º–∞—Ñ–∏—é üî´
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="9"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            Fall Guys
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="10"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            Among Us
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="11"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –æ—Ö–æ—Ç–∞ –Ω–∞ –∞—á–∏–≤–∫—É üéØ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="12"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∏–≥—Ä–æ–≤–æ–π –º–∞—Ä–∞—Ñ–æ–Ω üèÉ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="13"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Å–ø–∏–¥—Ä–∞–Ω
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="14"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –æ–±—É—á–µ–Ω–∏–µ –∏–≥—Ä–µ üß†
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="15"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Ñ—É–¥-—á–µ–ª–ª–µ–Ω–¥–∂ üçï
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="16"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Å–ø–µ—Ç—å/—Å—Ç–∞–Ω—Ü–µ–≤–∞—Ç—å/–Ω–∞—Ä–∏—Å–æ–≤–∞—Ç—å üé®
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="17"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∫—É–ª–∏–Ω–∞—Ä–Ω—ã–π —Å—Ç—Ä–∏–º üç≥
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="18"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            GeoGuessr
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="19"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –ø—Ä–æ–π—Ç–∏ –æ–Ω–ª–∞–π–Ω-—Ç–µ—Å—Ç—ã
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="20"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Ç—É—Ä–Ω–∏—Ä —Å –ø–æ–¥–ø–∏—Å—á–∏–∫–∞–º–∏ üèÜ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="21"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Ä–æ–∑—ã–≥—Ä—ã—à–∏
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="22"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            —Å–æ–≤–º–µ—Å—Ç–Ω–∞—è –∏–≥—Ä–∞ —Å–æ —Å—Ç—Ä–∏–º–µ—Ä–æ–º üëØ
                          </p>
                        </div>
                        <div
                          className="Templates__template"
                          key="23"
                          onClick={[Function]}
                        >
                          <p
                            className="Templates__title"
                          >
                            –∫–æ—Å–ø–ª–µ–π
                          </p>
                        </div>
                      </div>
                    </div>
                  </Templates>
                </Connect(Templates)>
              </div>
            </CreateChallenge>
          </Connect(CreateChallenge)>
        </Route>
      </withRouter(Connect(CreateChallenge))>
    </Provider>
  </Router>
</BrowserRouter>
`;
